### PR TITLE
Allow _ViewImports to remove default inherited tag helpers.

### LINKED ISSUE #2989: Unable to disable Url Resolution logic from _ViewImports.cshtml.
If a user wants to disable URL resolution (~/) they can do:

```
@removeTagHelper "*, Microsoft.AspNet.Mvc.Razor"
```

If done on a single page, everything works well. However, you're unable to add this `@removeTagHelper` in a `_ViewImports.cstml` to get the same effect.

### PR DESCRIPTION
**Pull request by pranavkm**
Additionally simplifying the contract for ChunkInheritaceUtility \ IChunkMerger so we don't have to understand the
order in which CodeTrees appear.

Fixes #2989

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @NTaylorMullen

File: src/Microsoft.AspNet.Mvc.Razor.Host/Directives/IChunkMerger.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/Directives/IChunkMerger.cs]: Interesting how the doc comment didn't have to be fully qualified before.

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/Directives/ChunkInheritanceUtilityTest.cs [hunk 2]
```diff
@@ -51,45 +51,44 @@ @inherits MyBaseType
...
-                    var viewImportsPath = PlatformNormalizer.NormalizePath(@"Views\_ViewImports.cshtml");
-                    Assert.Collection(chunkTreeResult.ChunkTree.Chunks,
+                        },
                         chunk =>
                         {
-                            Assert.IsType<LiteralChunk>(chunk);
+                            var setBaseTypeChunk = Assert.IsType<SetBaseTypeChunk>(chunk);
+                            Assert.Equal("MyBaseType", setBaseTypeChunk.TypeName);
                             Assert.Equal(viewImportsPath, chunk.Start.FilePath);
+
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/Directives/ChunkInheritanceUtilityTest.cs]: :boom:

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs [hunk 3]
```diff
@@ -119,10 +118,10 @@ public static TheoryData ViewImportsData
                     {
                         new[]
                         {
-                            CreateChunkTree(new TagHelperPrefixDirectiveChunk { Prefix = "THP1" }),
-                            CreateChunkTree(new AddTagHelperChunk { LookupText = "ATH" }),
-                            CreateChunkTree(new RemoveTagHelperChunk { LookupText = "RTH" }),
                             CreateChunkTree(new TagHelperPrefixDirectiveChunk { Prefix = "THP2" }),
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorParserTest.cs]: Ooo looks like we do merge taghelper prefixes correctly :+1:

File: test/WebSites/TagHelpersWebSite/Controllers/RemoveDefaultInheritedTagHelpersController.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc;
+
+namespace TagHelpersWebSite.Controllers
+{
+    public class RemoveDefaultInheritedTagHelpersController : Controller
+    {
+        public IActionResult Index() => View();
```
Comment (NTaylorMullen) [inline test/WebSites/TagHelpersWebSite/Controllers/RemoveDefaultInheritedTagHelpersController.cs]: lol

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**